corgidrp/calibrate_kgain.py

```diff
@@ -0,0 +1,1099 @@
+       plt.show() 
+    
+    # parameter for plot
+    parm1 = -0.5*np.log10(kgain)
+    
+    # Gaussian read noise value in DN
+    mean_rn_gauss_DN = np.mean(rn_gauss)
+    mean_rn_gauss_e = mean_rn_gauss_DN * kgain
+    
+    mean_rn_std_DN = np.mean(read_noise)
+    mean_rn_std_e = mean_rn_std_DN * kgain
+    
+    # If requested, plotting
+    if mkplot is not None:
+        # Create log-spaced averages for plotting
+        full_range_averages = np.logspace(log_plot1, log_plot2, log_plot3)
+        
+        plt.figure(num=3, figsize=(20, 10))
+        
+        # Main data plots
+        plt.plot(compiled_binned_averages, compiled_binned_total_deviations, 
+                 label='Shot Noise + Read Noise')
+        # Assuming `lower_bound` is 6
+        plt.plot(compiled_binned_averages, 
+                 compiled_binned_shot_deviations, label='Shot Noise')
+        plt.errorbar(10**logged_averages, 10**logged_deviations, 
+                     yerr=10**(logged_deviations_error), color='gray', 
+                     label='Fitted Points')
+        
+        # Extra lines
+        plt.plot(10**full_range_averages, 
+                 10**(full_range_averages * 0.50 + parm1), 
+                 'k-', label='Forced 0.50 Gradient')
+        
+        # Reference lines and text annotations
+        plt.axhline(y=mean_rn_std_DN, color='k', linestyle='--')
+        plt.axvline(x=95000, color='k', linestyle='--')
+        plt.text(2, 1.5, f'Fitted k gain = {np.round(kgain,1)} e/DN', fontsize=15)
+        plt.text(2, 10, 
+                 f'Read noise (Gaussian fit) = {np.round(mean_rn_gauss_DN,1)} DN = {mean_rn_gauss_e} e', 
+                 fontsize=15)
+        plt.text(2, 8, f'Read noise (Std) = {np.round(mean_rn_std_DN,1)} DN = {np.round(mean_rn_std_e,1)} e', 
+                 fontsize=15)
+        
+        # Scaling and labeling
+        plt.xscale('log')
+        plt.yscale('log')
+        plt.xlabel('Signal (DN)')
+        plt.ylabel('Noise (DN)')
+        plt.title('PTC curve fit')
+        plt.grid(True)
+        
+        # Setting axis limits
+        plt.xlim([1, 20000])
+        plt.ylim([1, 1000])
+        
+        # Legend and aesthetics
+        plt.legend(loc='upper left')
+        
+        plt.gca().tick_params(axis='both', which='major', labelsize=18)
+        
+        # Adjust figure position
+        plt.gcf().set_dpi(100)
+        plt.gcf().set_size_inches(20, 10)
+        
+        # Display the plot
+        plt.show()
+    
+    # prepare PTC array
+    ptc_list = [compiled_binned_averages, 
+             compiled_binned_shot_deviations]
+    ptc = np.column_stack(ptc_list)
+
+    prhd, exthd = create_default_headers()
+    gain_value = np.array([[kgain]])
+    kgain = data.KGain(gain_value, pri_hdr = prhd, ext_hdr = exthd)
+    
+    return (kgain, mean_rn_gauss_e, mean_rn_std_e, ptc)
+    
+
+
+"""
+Class to hold input-checking functions to minimize repetition
```

Comment:
FYI, I put this same class in `check.py` in a `util` folder and `test_check.py` in the `tests` folder on my PR since the calibration code in that branch also uses these checks.

Reply:
@maxwellm: There are two "check" py functions: calibration and tests. It seems that we should have both. 

@kjl0025: which is your PR (link)? I'd like to find the util foder with check.py and test_check.py

Reply:
Found it: https://github.com/roman-corgi/corgidrp/pull/105/files

Reply:
I'm not quite sure I understand why we need two different check.py files

Reply:
test_check.py is the unit test of check.py.